WordPress/Sniffs/WP/DeprecatedParametersSniff.php

```diff
@@ -0,0 +1,349 @@
+			2 => array(
+				'value'   => '',
+				'version' => '2.8.1',
+			),
+		),
+		'the_attachment_link' => array(
+			3 => array(
+				'value'   => false,
+				'version' => '2.5.0',
+			),
+		),
+		'the_author' => array(
+			1 => array(
+				'value'   => '',
+				'version' => '2.1.0',
+			),
+			2 => array(
+				'value'   => true,
+				'version' => '1.5.0',
+			),
+		),
+		'the_author_posts_link' => array(
+			1 => array(
+				'value'   => '',
+				'version' => '2.1.0',
+			),
+		),
+		'trackback_rdf' => array(
+			1 => array(
+				'value'   => '',
+				'version' => '2.5.0',
+			),
+		),
+		'trackback_url' => array(
+			1 => array(
+				'value'   => true,
+				'version' => '2.5.0',
+			),
+		),
+		'update_blog_option' => array(
+			4 => array(
+				'value'   => null,
+				'version' => '3.1.0',
+			),
+		),
+		'update_blog_status' => array(
+			4 => array(
+				'value'   => null,
+				'version' => '3.1.0',
+			),
+		),
+		'update_user_status' => array(
+			4 => array(
+				'value'   => null,
+				'version' => '3.0.2',
+			),
+		),
+		'unregister_setting' => array(
+			4 => array(
+				'value'   => '',
+				'version' => '4.7.0',
+			),
+		),
+		'wp_get_http_headers' => array(
+			2 => array(
+				'value'   => false,
+				'version' => '2.7',
+			),
+		),
+		'wp_get_sidebars_widgets' => array(
+			1 => array(
+				'value'   => true,
+				'version' => '2.8.1',
+			),
+		),
+		'wp_install' => array(
+			5 => array(
+				'value'   => '',
+				'version' => '2.6.0',
+			),
+		),
+		'wp_new_user_notification' => array(
+			2 => array(
+				'value'   => null,
+				'version' => '4.3.1',
+			),
+		),
+		'wp_notify_postauthor' => array(
+			2 => array(
+				'value'   => null,
+				'version' => '3.8.0',
+			),
+		),
+		'wp_title_rss' => array(
+			1 => array(
+				'value'   => '&#8211;',
+				'version' => '4.4.0',
+			),
+		),
+		'wp_upload_bits' => array(
+			2 => array(
+				'value'   => '',
+				'version' => '2.0.0',
+			),
+		),
+		'xfn_check' => array(
+			3 => array(
+				'value'   => '',
+				'version' => '2.5.0',
+			),
+		),
+
+	); // End $target_functions.
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.12.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$paramCount = count( $parameters );
+		foreach ( $this->target_functions[ $matched_content ] as $position => $parameter_args ) {
+
+			// Check that number of parameters defined is not less than the position to check.
+			if ( $position > $paramCount ) {
+				break;
+			}
+
+			// The list will need to updated if the default value is not supported.
```

Comment:
@grappler Didn't you remove this condition earlier based on a previous comment I made ? It's superfluous. (And removing it would sort the comment out as well ;-) )